numpy_financial/_financial.py

```diff
@@ -767,28 +741,20 @@ def irr(values):
     if values.ndim != 1:
         raise ValueError("Cashflows must be a rank-1 array")
 
-    # Strip leading and trailing zeros. Since we only care about
-    # positive roots we can neglect roots at zero.
-    non_zero = np.nonzero(np.ravel(values))[0]
-    values = values[int(non_zero[0]):int(non_zero[-1])+1]
+    # NPV = V0 * (1+eirr)^0 + V1 * (1+eirr)^-1 + ...
+    #  let x = 1 / (1+eirr)
+    # NPV = V0 * x^0        + V1 * x^1         + ...
```

Comment:
```suggestion
# We aim to solve eirr such that NPV is exactly zero. This can be framed as
    # simply finding the closest root of a polynomial to a given initial guess
    # as follows:
    #           V0           V1           V2           V3
    # NPV = ---------- + ---------- + ---------- + ---------- + ...
    #       (1+eirr)^0   (1+eirr)^1   (1+eirr)^2   (1+eirr)^3
    #
    # by letting x = 1 / (1+eirr), we substitute to get
    #
    # NPV = V0 * x^0   + V1 * x^1   +  V2 * x^2  +  V3 * x^3  + ...
    # 
    # which we solve using Newton-Raphson and then reverse out the solution 
    # as eirr = 1/x - 1 (if we are close enough to a solution)
```

Reply:
I think it's more concise to leave it as is. At this point I expect anyone looking at the source code to know this method, given that we already have some comments explaining it. Maybe it would be better to spread them out throughout the code a bit more?

P.S. I'm at work right now so I'll probably only get back to this tonight

Reply:
up to you, i've become inclined over the years to over-explain to make it easier for new-comers, but i don't really know what the existing level of commenting in the codebase is, so maintainers call - I've added a couple of shorter suggestions below instead if you want to use.

Reply:
> i've become inclined over the years to over-explain to make it easier for new-comers

I agree. I guess that I've become too used to seeing the code myself so it's easier for me to read into it.

> but i don't really know what the existing level of commenting in the codebase is

Currently there are very few comments, and those that do exist are very specific to what that local code snippet is doing.

> so maintainers call

While I've been active lately numpy-financial in general is poorly maintained. In fact, it was split into it's own module because the current numpy maintainers did **not** want to maintain them in the numpy namespace.

> I've added a couple of shorter suggestions below instead if you want to use.

Thanks. I'll look over this more when I get back from work. And make a decision.

Reply:
I'm going to merge your more verbose version. I think it is a good addition ;) and will save whoever works on this later (probably me) some time. If future maintainers find it to long winded it may always be removed. Thank you for taking the time to contribute.